### PR TITLE
CS-43630 - Handling the embedded entries cretion in RTE, version bump

### PR DESCRIPTION
**Pull request by cs-raj**
version bump:
@contentstack/cli : 1.16.0 -> 1.17.0
@contentstack/cli-cm-import: 1.14.2 -> 1.15.0
@contentstack/cli-cm-clone: 1.10.2 -> 1.10.3
@contentstack/cli-cm-seed: 1.7.2 -> 1.7.3

### CODE REVIEW

File: packages/contentstack-import/src/utils/entries-helper.ts [hunk 1]
```diff
@@ -534,11 +550,37 @@ export const restoreJsonRteEntryRefs = (
...
       }
+      case 'text': {
+        if (entry[element.uid] && element.field_metadata.rich_text_type) {
+          entry[element.uid] = sourceStackEntry[element.uid];
+          const matches: string[] = [];
+          Object.keys(uidMapper).forEach((uid) => {
+            if (sourceStackEntry[element.uid].indexOf(uid) !== -1) {
+              matches.push(uid);
+            }
+          });
```
Comment (aman19K) [inline packages/contentstack-import/src/utils/entries-helper.ts:561]: @cs-raj here can we use filter? For example:-

const matches = Object.keys(uidMapper).filter((uid) =>  if (sourceStackEntry[element.uid].indexOf(uid) !== -1) return uid;)

File: packages/contentstack-import/src/utils/entries-helper.ts [hunk 2]
```diff
@@ -534,11 +550,37 @@ export const restoreJsonRteEntryRefs = (
...
+      }
     }
   }
   return entry;
 };
 
+function updateUids(str:string,match:string,uidMapper:Record<string, string>) {
+  return str.replace(new RegExp(match, 'g'), (match:string) => uidMapper[match])
+}
+
```
Comment (aman19K) [inline packages/contentstack-import/src/utils/entries-helper.ts:583]: @cs-raj formatting missing

File: packages/contentstack-import/src/import/modules/entries.ts [hunk 3]
```diff
@@ -56,6 +56,8 @@ export default class EntriesImport extends BaseClass {
   private autoCreatedEntries: Record<string, any>[];
   private taxonomiesPath: string;
   public taxonomies: Record<string, unknown>;
+  public rteCTs: any;
+  public rteCTsWithRef: any;
```
Comment (aman19K) [inline packages/contentstack-import/src/import/modules/entries.ts:60]: @cs-raj private will be fine here?

File: packages/contentstack-import/src/utils/entries-helper.ts (hunk 1 above)
Comment (cs-raj) [reply packages/contentstack-import/src/utils/entries-helper.ts]: Added

File: packages/contentstack-import/src/utils/entries-helper.ts (hunk 2 above)
Comment (cs-raj) [reply packages/contentstack-import/src/utils/entries-helper.ts]: Formatted

File: packages/contentstack-import/src/import/modules/entries.ts (hunk 3 above)
Comment (cs-raj) [reply packages/contentstack-import/src/import/modules/entries.ts:60]: While adding the test cases, It will be directly accessible

**Review by aman19K — APPROVED** [review]

**Review by antonyagustine — APPROVED** [review]

**Pull request merged**